remotespark/utils/decorators.py

```diff
@@ -0,0 +1,17 @@
+from remotespark.utils import utils
+
+
+def add_guid(attr_name):
+    def class_decorator(cls):
+        def get_attr(self, attr=attr_name):
+            return getattr(self, "_" + attr)
+
+        def set_attr(self, value, attr=attr_name):
+            setattr(self, "_" + attr, value)
+
+        prop = property(get_attr, set_attr)
+        setattr(cls, attr_name, prop)
+        setattr(cls, "_" + attr_name, utils.generate_uuid())  # Default value for that attribute
```

Comment:
This adds the same GUID to all sessions (as a class variable).  Two alternatives:

* Edit the class's `__init__` method directly in this decorator

* Eschew the decorator and just use a base class for all things with GUID's in them (I prefer this)